### PR TITLE
fix amalgamated build

### PR DESCRIPTION
**Pull request by jamesfAnet**

### CODE REVIEW

**Comment by bkaradzic** [comment]
Why `GetTempPathA` doesn't have the issue?

File: src/filepath.cpp [hunk 1]
```diff
@@ -16,7 +16,9 @@
 #endif // !BX_CRT_NONE
 
 #if BX_PLATFORM_WINDOWS
+#if !defined(GetModuleFileName)
 extern "C" __declspec(dllimport) unsigned long __stdcall GetModuleFileNameA(void* _module, char* _outFilePath, unsigned long _size);
+#endif
```
Comment (jamesfAnet) [inline src/filepath.cpp:21]: when in amalgamated build, we do include order of 

file.cpp
filepath.cpp

file.cpp includes dirent.h, which includes windows.h, which eventually includes libloaderapi.h which defines this function GetModuleFileNameA.

You can repro the compilation problem if you add an include of dirent.h to the top of this cpp file. 

In my fork I'm adding an ifndef like this to work around it.

Comment (jamesfAnet) [reply src/filepath.cpp:21]: compile error for reference
```
1>bx\src\filepath.cpp(20,58): error C2733: 'GetModuleFileNameA': you cannot overload a function with 'extern "C"' linkage
1>bx\src\filepath.cpp(20,58): error C2733: extern "C" __declspec(dllimport) unsigned long __stdcall GetModuleFileNameA(void* _module, char* _outFilePath, unsigned long _size);
1>bx\src\filepath.cpp(20,58): error C2733:                                                          ^
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um\libloaderapi.h(208,1): message : see declaration of 'GetModuleFileNameA'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um\libloaderapi.h(208,1): message : GetModuleFileNameA(
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um\libloaderapi.h(208,1): message : ^
```

Comment (bkaradzic) [reply src/filepath.cpp:21]: Ah ok.

**Pull request merged**